operator/pkg/util/merge_iop.go

```diff
@@ -0,0 +1,176 @@
+	ClusterResources       *protobuf.BoolValue              `json:"clusterResources" patchStrategy:"merge"`
+	IstioCni               *v1alpha12.CNIConfig             `json:"istio_cni" patchStrategy:"merge"`
+	MeshConfig             *meshConfig                      `json:"meshConfig" patchStrategy:"merge"`
+	Base                   *v1alpha12.BaseConfig            `json:"base" patchStrategy:"merge"`
+	IstiodRemote           *v1alpha12.IstiodRemoteConfig    `json:"istiodRemote" patchStrategy:"merge"`
+}
+
+type gatewaysConfig struct {
+	IstioEgressgateway  *egressGatewayConfig  `json:"istio-egressgateway" patchStrategy:"merge"`
+	IstioIngressgateway *ingressGatewayConfig `json:"istio-ingressgateway" patchStrategy:"merge"`
+}
+
+// Configuration for an ingress gateway.
+type ingressGatewayConfig struct {
+	Cpu                *v1alpha12.CPUTargetUtilizationConfig `json:"cpu" patchStrategy:"merge"`
+	MeshExpansionPorts []*v1alpha12.PortsConfig              `json:"meshExpansionPorts" patchStrategy:"merge"`
+	Ports              []*v1alpha12.PortsConfig              `json:"ports" patchStrategy:"merge"`
+	SecretVolumes      []*v1alpha12.SecretVolume             `json:"secretVolumes" patchStrategy:"merge"`
+	Zvpn               *v1alpha12.IngressGatewayZvpnConfig   `json:"zvpn" patchStrategy:"merge"`
+}
+
+type egressGatewayConfig struct {
+	Ports         []*v1alpha12.PortsConfig  `json:"ports" patchStrategy:"merge"`
+	Resources     *v1alpha12.Resources      `json:"resources" patchStrategy:"merge"`
+	SecretVolumes []*v1alpha12.SecretVolume `json:"secretVolumes" patchStrategy:"merge"`
+	Zvpn          *v1alpha12.ZeroVPNConfig  `json:"zvpn" patchStrategy:"merge"`
+}
+
+type meshConfig struct {
+	TcpKeepalive                   *v1alpha3.ConnectionPoolSettings_TCPSettings_TcpKeepalive `json:"tcpKeepalive" patchStrategy:"merge"`
+	DefaultConfig                  *v1alpha13.ProxyConfig                                    `json:"defaultConfig" patchStrategy:"merge"`
+	ConfigSources                  []*v1alpha13.ConfigSource                                 `json:"configSources" patchStrategy:"merge"`
+	TrustDomainAliases             []string                                                  `json:"trustDomainAliases" patchStrategy:"merge"`
+	DefaultServiceExportTo         []string                                                  `json:"defaultServiceExportTo" patchStrategy:"merge"`
+	DefaultVirtualServiceExportTo  []string                                                  `json:"defaultVirtualServiceExportTo" patchStrategy:"merge"`
+	DefaultDestinationRuleExportTo []string                                                  `json:"defaultDestinationRuleExportTo" patchStrategy:"merge"`
+	LocalityLbSetting              *v1alpha3.LocalityLoadBalancerSetting                     `json:"localityLbSetting" patchStrategy:"merge"`
+	Certificates                   []*v1alpha13.Certificate                                  `json:"certificates" patchStrategy:"merge"`
```

Comment:
does this mean we cannot replace these lists?

Reply:
This seems not desirable - but also not a big deal since we don't have defaults anyways...